### PR TITLE
Enable preview options for multiple selected nodes

### PR DESCRIPTION
**Pull request by Benglin**
Background
-----
This pull request implements the following new canvas menu options:

1. Lacing
2. Show/Hide all geometry preview
3. Show/Hide upstream geometry preview

These options show up when user right-clicks on the canvas, and they apply the setting to all the nodes currently selected. When there is no selection, these canvas menu options are disabled.

canvas-menu-multiple-nodes

The following options on contextual node menu (one that shows up when user right-clicks on a node) continues to work the same way they do today -- they affect only the node that is being right-clicked:

1. Lacing
2. Preview
3. Preview Upstream

New Behaviors
-----

1. Lacing: The `Lacing` sub-menu has a check box right next to the current active `LacingStrategy` value. This value is derived from the `NodeModel.ArgumentLacing` values of nodes in selection. For example, if more nodes in selection are having `ArgumentLacing` as `CrossProduct`, then the sub-menu item `CrossProduct` will be checked.

2. Show/Hide all geometry preview: If any of the nodes in selection has its `IsVisible` property set to `true`, then this menu item will be shown as *Hide all geometry preview*. Otherwise, it will be shown as *Show all geometry preview*.

3. Show/Hide upstream geometry preview: If any of the nodes in selection has its `IsUpstreamVisible` property set to `true`, then this menu item will be shown as *Hide upstream geometry preview*. Otherwise, it will be shown as *Show upstream geometry preview*.

Notes to Reviewer
-----
All the tests are passing/failing as expected. Hi @nguyen-binh-minh, please have a look. Thanks!

### CODE REVIEW

File: src/DynamoCoreWpf/Commands/WorkspaceCommands.cs [hunk 1]
```diff
@@ -11,11 +17,20 @@ public partial class WorkspaceViewModel
         private DelegateCommand _resetFitViewToggleCommand;
         private DelegateCommand _findByIdCommand;
         private DelegateCommand _alignSelectedCommand;
+        private DelegateCommand _enableNodePreviewCommand;
+        private DelegateCommand _enableUpstreamPreviewCommand;
+        private DelegateCommand _setArgumentLacingCommand;
```
Comment (Benglin) [inline src/DynamoCoreWpf/Commands/WorkspaceCommands.cs]: Hi @nguyen-binh-minh, I don't think the above two `_enableXyzCommand` are used, can you remove them?

File: src/DynamoCoreWpf/Commands/WorkspaceCommands.cs [hunk 2]
```diff
@@ -157,5 +184,80 @@ public DelegateCommand UnPauseVisualizationManagerCommand
...
+        public bool AnyNodeUpstreamVisible
+        {
+            get
+            {
+                return DynamoSelection.Instance.Selection.
+                    OfType<NodeModel>().Any(n => n.IsUpstreamVisible);
+            }
+        }
+
+        public string ShowHideTextBubble
```
Comment (nguyen-binh-minh) [inline src/DynamoCoreWpf/Commands/WorkspaceCommands.cs]: @Benglin this property is not used in this user story but for another one.
Should we remove this?

Comment (Benglin) [inline src/DynamoCoreWpf/Commands/WorkspaceCommands.cs]: Oh yes, good point. Please remove it if it does not belong here. Thanks!

**Comment by nguyen-binh-minh** [comment]
@Benglin LGTM

**Comment by Benglin** [comment]
Thanks again for your help and review, @nguyen-binh-minh. I will merge this in tomorrow first thing.

**Pull request merged**